frontend/lib/src/components/widgets/DataFrame/columns/LinkColumn.ts

```diff
@@ -104,35 +103,45 @@ function LinkColumn(props: BaseColumnProps): BaseColumn {
     sortMode: "default",
     validateInput,
     getCell(data?: any, validate?: boolean): GridCell {
+      if (!data) {
+        return {
+          ...cellTemplate,
+          isMissingValue: true,
+        } as LinkCell
+      }
+
+      const href: string = data
       if (typeof validateRegex === "string") {
         // The regex is invalid, we return an error to indicate this
         // to the developer:
-        return getErrorCell(toSafeString(data), validateRegex)
+        return getErrorCell(toSafeString(href), validateRegex)
       }
 
       if (validate) {
-        const validationResult = validateInput(data)
+        const validationResult = validateInput(href)
         if (validationResult === false) {
           // The input is invalid, we return an error cell which will
           // prevent this cell to be inserted into the table.
           // This cell should never be actually displayed to the user.
           // It's mostly used internally to prevent invalid input to be
           // inserted into the table.
-          return getErrorCell(toSafeString(data), "Invalid input.")
-        } else if (typeof validationResult === "string") {
-          // Apply corrections:
-          data = validationResult
+          return getErrorCell(toSafeString(href), "Invalid input.")
         }
       }
 
       return {
         ...cellTemplate,
-        data: notNullOrUndefined(data) ? toSafeString(data) : null,
-        isMissingValue: isNullOrUndefined(data),
-      } as UriCell
+        data: {
+          kind: "link-cell",
+          href: href,
+          displayText: getLinkDisplayValue(href, parameters.display_text),
+        },
+        copyData: href,
```

Comment:
nit: to get the pointer mouse for the full cell, you can just add this here:

```suggestion
        copyData: href,
        cursor: "pointer",
```

This isn't perfect since it would show the pointer on the full cell and not just the link, but it feels better than not showing the pointer at all.

Reply:
Yeah, I could not see a way to conditionally update the cursor property of the cell when `isHovered` was true, since it is readonly.